### PR TITLE
Br/test reorg.w 7656001

### PR DESCRIPTION
**Pull request by bredmond-sf**
Arranges tests to live with source.
Separates resource tests into separate suite
Updates contributing doc.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -53,6 +66,7 @@ Profiles should be named with clarity to be descriptive of the purpose of that p
 #### Testing 
 
 We have provided a test bed for ensuring your custom validations work as expected.  We have profiled a [utils](../test/utils.js) file that provides some utility methods to easily test your profile.  Your test file should have the same name as your profile to clearly relate tests to profiles.
+Test files live alongside the source files. Unittests for `src/commandName/commandName.ts` should have `src/commandName/commandName.test.ts` or for more complex test suites, `src/commandName/commandName.functionName.test.ts`. Resources should similiarly have tests in the `resources` path. All tests will be run in the CI. They will not be included when published.
```
Comment (unandyala) [inline .github/CONTRIBUTING.md:69]: space between unit and tests

File: src/diffTool/rulesProcessor.test.ts [hunk 2]
```diff
@@ -4,43 +4,30 @@
...
-import sinon from "sinon";
-import { ramlToolLogger } from "../../src";
 import chaiAsPromised from "chai-as-promised";
 
 /* eslint-disable @typescript-eslint/no-use-before-define*/
 
 const expect = chai.expect;
 chai.use(chaiAsPromised);
 
-let loggerSpy;
```
Comment (unandyala) [inline src/diffTool/rulesProcessor.test.ts:21]: what is the issue here? Is it related to the log level

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -17,6 +18,18 @@
...
 - [FAQ](#faq)
 
+## Repository Structure
+
+`bin/**` - default oclif wrappers generated by oclif. Should not be directly edited.
+`src/**` - files to transpiled by tsc to the lib folder using `npm run compile`.
+`src/index.ts` - contains package exports. Becomes lib/index.js and is the entrypoint.
+`src/common/*.ts` - functions that are shared by more than one command such as logger and RAML parser.
+`src/{commandName}/` - folders, named after commands whenever possible, to contain logic needed for commands and index.ts for oclif.
+`src/commands/*.ts` - oclif requires a particular file structure to find commands. It needs a folder configured in the package.json where there is a filename for each command. This folder contains one name per command that exports the matching command.
```
[the remark this answers is not in the document]
Comment (bredmond-sf) [reply .github/CONTRIBUTING.md:28]: Yes, indeed, fixing.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -17,6 +18,18 @@
 - [Release Process and Cadence](#release-process-and-cadence)
 - [FAQ](#faq)
 
+## Repository Structure
+
+`bin/**` - default oclif wrappers generated by oclif. Should not be directly edited.
+`src/**` - files to transpiled by tsc to the lib folder using `npm run compile`.
```
[the remark this answers is not in the document]
Comment (bredmond-sf) [reply .github/CONTRIBUTING.md:24]: fixing

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (bredmond-sf) [reply .github/CONTRIBUTING.md:69]: fixing

File: src/diffTool/rulesProcessor.test.ts (hunk 2 above)
Comment (bredmond-sf) [reply src/diffTool/rulesProcessor.test.ts:21]: When the last test in the logger test file (which switches the loglevel to debug) is included in the run, it causes the spy in this test to fail. I tried a bunch of things to get it to work, like resetting the level in the logger test and explicitly setting the level here before creating the spy, but nothing worked.

File: src/download/exchangeDownloader.test.ts [hunk 5]
```diff
@@ -23,13 +23,13 @@ import nock from "nock";
...
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-const getAssetWithVersion = require("./resources/getAssetWithVersion");
+const getAssetWithVersion = require("../../test/download/resources/getAssetWithVersion");
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-const getAssetWithoutVersion = require("./resources/getAsset");
+const getAssetWithoutVersion = require("../../test/download/resources/getAsset");
 
 before(() => {
   chai.should();
```
Comment (wjhsf) [inline src/download/exchangeDownloader.test.ts:35]:
```suggestion
```

File: tsconfig.json [hunk 6]
```diff
@@ -23,15 +23,12 @@
...
     "exclude": [
-        "node_modules",
-        "release",
-        "test",
-        "gulpfile.ts"
+        "**/*.test.ts"
     ],
     "include": [
-        "src"
+        "src/**/*.ts"
```
Comment (wjhsf) [inline tsconfig.json:32]: Why this change?

File: package.json [hunk 7]
```diff
@@ -122,8 +122,11 @@
     "prepack": "npm run clean && npm run compile && oclif-dev manifest && oclif-dev readme",
     "pretest": "npm run depcheck && npm run lint",
     "lint": "eslint \"**/*.ts\" --quiet --fix",
-    "test:debug": "TS_NODE_COMPILER_OPTIONS='{\"rootDir\":\".\"}' nyc mocha \"test/**/*.test.ts\"",
-    "test": "TS_NODE_COMPILER_OPTIONS='{\"rootDir\":\".\"}' nyc mocha \"test/**/*.test.ts\" --reporter=xunit --reporter-options output=reports/generator.xml",
+    "testrunner": "TS_NODE_COMPILER_OPTIONS='{\"rootDir\":\".\"}' mocha",
```
Comment (wjhsf) [inline package.json:125]: Is there any way to configure a tsconfig for the tests directory so that this isn't necessary?

Comment (bredmond-sf) [reply package.json]: fixed

File: tsconfig.json (hunk 6 above)
Comment (bredmond-sf) [reply tsconfig.json]: removed

File: src/download/exchangeDownloader.test.ts (hunk 5 above)
Comment (bredmond-sf) [reply src/download/exchangeDownloader.test.ts]: done
